Yolo model evaluation

punithsekar opened the pull request:
### Problem description
Evaluate yolo models.(mAPval 50-95)

### What's changed
Added evaluation code for yolov8x, yolo11n, and yolov4. 
The mAPval 50-95 for the following models are: (for 500 samples)
 YOLOv4(320x320 resolution) - 0.7547

Currently, commented out the work of yolov8x and yolo11n as they are not merged to main.
AI's :- 
- To add the evaluation for other yolo models once its done.

### Checklist
- [ ] [All post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml) CI passes

punithsekar committed:
#0: yolo evaluation

mbahnasTT commented:
Can you add a version from the test that is running on the CPU model?
Can we add a larger number of samples, like 500?

mbahnasTT commented:
Saichand, No! I need what I requested before approving the PR.

ssinghalTT commented:
@saichandax , also mention the dataset used. Ideally the mAP values should come from the complete validation dataset of datasets like COCO

punithsekar commented:
Sure @mbahnasTT @ssinghalTT . I will update on the concerns you mentioned.

punithsekar commented:
@mbahnasTT @ssinghalTT , 
I have added the test for 500 samples and updated the numbers in the readme.
The dataset which we use is COCO 2017 which I added in the readme.

Mohamed,  Here we are comparing between the tt model and ground truth(dataset values) values. We are not running the torch version currently. But we can compare the mAP of torch version vs groundtruth(dataset values) by updating the model_type= "torch_model" in fixtures.

punithsekar commented:
We cannot merge the PR now,

- Once dependent Yolo models are pushed to the main we can merge this PR to the main.
- Need to automate weights download of dependent models, This will be done once the dependent models merge to main.

Once above checklist is done we can merge this PR.

tt-rkim commented on models/experimental/yolo_evaluation/yolo_common_evaluation.py:
```diff
@@ -0,0 +1,582 @@
+    source_list = [i["filepath"] for i in dataset]
+    data_set = LoadImages(path=[i["filepath"] for i in dataset])
+
+    with open("/home/ubuntu/fiftyone/coco-2017/info.json", "r") as file:
```
Hardcoding this here? What's going on?

punithsekar replied:
I will update it!

tt-rkim commented on models/experimental/yolo_evaluation/README.md:
```diff
@@ -0,0 +1,22 @@
+# Yolov evaluation
+
+- Using `coco-2017` validation dataset.
+- Loading the dataset using `fiftyone` package. Our tt-metal dosen't have fiftyone, Download the package using `pip install fiftyone`.
```
Why mention this if you put it in `requirements-dev.txt`? Please replace such suggestions with a more generic suggestion, such as using `create_venv.sh` or ensuring that users have the libraries stated in `requirements-dev.txt`.

tt-rkim reviewed (commented):
Why do we have another perf test for yolo models? This must be like the 5th one I've seen. Can I get an explanation? I will further ping people on slack.

punithsekar commented:
> Why do we have another perf test for yolo models? This must be like the 5th one I've seen. Can I get an explanation? I will further ping people on slack.

The intention of this PR is to have a common evaluation file for all Yolo models.

mbahnasTT commented:
@tt-rkim the yolo model bring up effort is coordinated with @mywoodstock. My team is owning the first bring up of CNN models as a whole, and we work closely with Conv team, as well as the other OP teams.
We are currently bringing up almost of versions from yolo, starting v4 till v12.

mbahnasTT commented:
@punithsekar 
- Please, can you add the exact command of running the torch model in the readme file? like adding the full script, instead of explaining what needs to be done.
- Yolov4 (with post-processing module on device) is merged into main now, so you can just enable that check, and add a `skip` option for the other 2 models, till they are added to main.

mbahnasTT commented:
@tt-rkim For the time being we don't need to add in CI, as it's mainly sort of demo on the model inference accuracy, and should be covered by the PCC checks. 
However, we may need to add in the future, if PCC checks are not enough to capture any regression on these scores.

tt-rkim commented:
OK sounds good

Please report on any increases in docker image size from the PR gate if so.
